### DISCUSSION TITLE
Firebase phone auth not initiating OTP only in release mode but works on Debug mode even after added sha1 for release mode

### DISCUSSION THREAD

**Discussion by MuthuSubramaniam** [firebase/flutterfire#3401] [Q&A] [answered]
Steps how I implemented.

VERSION:   firebase_auth: ^0.18.0+1

As advised, I implemented all the way. It was quite good in debug mode. Unfortunately its not working in release mode even I added my debug and release sha1 keys in firebase console.

Step 1. Coding part is here

              final PhoneVerificationCompleted verificationCompleted =
                     (PhoneAuthCredential user) {
                  successMessage('Your phone number verified successfully!');
                 _loginNotifier.validate();
              };
             final PhoneVerificationFailed verificationFailed = (FirebaseAuthException authException) {
                  errorMessage('Phone number verification failed. Code: ${authException.code}. Message: 
                 ${authException.message}');
            };

             final PhoneCodeAutoRetrievalTimeout codeAutoRetrievalTimeout =
                    (String verificationId) {
                 this._verificationId = verificationId;
                print("time out");
             };

             firebaseAuth = FirebaseAuth.instance;
             firebaseAuth.verifyPhoneNumber(
                   phoneNumber: '+$_phoneWithCC',
                   timeout: const Duration(seconds: 60),
                   verificationCompleted: verificationCompleted,
                   verificationFailed: verificationFailed,
                   codeSent: codeSent,
                   codeAutoRetrievalTimeout: codeAutoRetrievalTimeout);

Step 2: Added sha1 key's in console
![image](https://user-images.githubusercontent.com/39517706/91658581-fa954980-eafb-11ea-9b15-e950beb91155.png)

**Error I am getting:**

As I said, in debug mode all good as expected But in release mode, after initiating the otp function, immediately PhineVerificationFailed with null message and null code.

If still anything I missed please advice. Thanks all

**Comment by Sreejithtechno** [comment]
same error

**Reply by MuthuSubramaniam** [reply]
@Sreejithtechno any update?

**Comment by alvinvin00** [comment, marked as answer]
did you update your google-services.json after adding those keys? If you haven't go update the JSON and rebuild

**Reply by MuthuSubramaniam** [reply]
yes its up to date

**Reply by alvinvin00** [reply]
I remember i encountered this problem before. Try adding the SHA256 hash on Firebase Console, don’t forget to update your google_services.json on your project after you added them.

**Reply by hkodes** [reply]
I did everything. Still it's the same in Release Mode but wrks in Debug. What am I missing here? 
@MuthuSubramaniam  Did you solve it?

**Reply by MuthuSubramaniam** [reply]
yes same for me.. still same.. so i went with sms provider.
